<launch>
    <arg name="subject" />
    <arg name="head_pcs_dir" />
    <arg name="ell_regs_dir" />
    <arg name="kinect_points" />

    <node pkg="hrl_head_registration" name="r_head_registration_srv" type="head_registration_srv"
          output="screen" args="$(arg head_pcs_dir)/$(arg subject)_r_face.bag">
        <param name="max_iters" value="300" />
        <param name="color_weight" value="0.0" />
        <param name="icp_max_corresp" value="0.3" />
        <param name="icp_trans_eps" value="0.0000000001" />
        <param name="trim_radius" value="0.13" />
        <param name="model_radius" value="0.03" />
        <param name="color_std_thresh" value="4.0" />
        <param name="hue_weight" value="5.0" />
        <remap from="/head_registration" to="/head_registration_r" />
        <remap from="/points" to="$(arg kinect_points)" />
    </node>
    <node pkg="hrl_head_registration" name="l_head_registration_srv" type="head_registration_srv"
          output="screen" args="$(arg head_pcs_dir)/$(arg subject)_l_face.bag">
        <param name="max_iters" value="300" />
        <param name="color_weight" value="0.0" />
        <param name="icp_max_corresp" value="0.3" />
        <param name="icp_trans_eps" value="0.0000000001" />
        <param name="trim_radius" value="0.13" />
        <param name="model_radius" value="0.03" />
        <param name="color_std_thresh" value="4.0" />
        <param name="hue_weight" value="5.0" />
        <remap from="/head_registration" to="/head_registration_l" />
        <remap from="/points" to="$(arg kinect_points)" />
    </node>
    <node pkg="hrl_face_adls" name="registration_loader" type="registration_loader.py"
          output="screen" >
        <remap from="/feedback" to="/wt_log_out" />
        <param name="registration_prefix" value="$(arg ell_regs_dir)/$(arg subject)" />
        <rosparam>
            registration_files:
                shaving:
                    r: "_shaver_r_ellipsoid_registration.bag"
                    l: "_shaver_l_ellipsoid_registration.bag"
                scratching:
                    r: "_scratcher_r_ellipsoid_registration.bag"
                    l: "_scratcher_l_ellipsoid_registration.bag"
                feeding:
                    r: "_spoon_r_ellipsoid_registration.bag"
                    l: "_spoon_l_ellipsoid_registration.bag"
        </rosparam>
    </node>
    <node pkg="hrl_head_registration" name="head_alignment_confirm" type="head_alignment_confirm"
          output="screen" respawn="True" />
</launch>
